tensorboard/webapp/plugins/plugins.component.html

```diff
@@ -14,4 +14,12 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 -->
-<div #plugins class="plugins"></div>
+<div
+  #pluginContainer
+  class="plugins"
+  [ngSwitch]="activePlugin?.loading_mechanism.type"
```

Comment:
To make sure I understand, when Angular-based built-in plugins are added in the future, HTML code like the following will need to be added in this div? (Just trying to understand how this works.)

Reply:
Correct. We are using Angular template system to build things AOT and render it with other angular components (as opposed to `document.createElement` which I think we should avoid). Btw, there is very weak argument about the dependency graph because the NgModule already has import statement to pull in, for example, debugger.